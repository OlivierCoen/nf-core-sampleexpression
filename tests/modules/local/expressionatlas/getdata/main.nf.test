nextflow_process {

    name "Test Process EXPRESSIONATLAS_GETDATA"
    script "modules/local/expressionatlas/getdata/main.nf"
    process "EXPRESSIONATLAS_GETDATA"
    tag "getdata"

    test("Transcriptome Analysis of the potato (rnaseq)") {

        tag "getdata_potato"

        when {

            process {
                """
                input[0] = "E-MTAB-552"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Transcription profiling by array of Arabidopsis mutant for fis2 (microarray)") {

        tag "getdata_arabido"

        when {

            process {
                """
                input[0] = "E-TABM-1007"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
