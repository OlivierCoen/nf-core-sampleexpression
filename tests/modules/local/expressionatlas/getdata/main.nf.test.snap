{
    "Transcriptome Analysis of the potato": {
        "content": [
            {
                "0": [
                    "E-MTAB-552.rnaseq.design.csv:md5,142d6bb20d784ba414fa76a6e73bb37f"
                ],
                "1": [
                    "E-MTAB-552.rnaseq.raw.csv:md5,315f6ab694cb809bd33f481ad326c5a7"
                ],
                "2": [
                    
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "R",
                        "[1] \"R version 4.3.3 (2024-02-29)\""
                    ]
                ],
                "4": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "ExpressionAtlas",
                        "[1] \u20181.30.0\u2019"
                    ]
                ],
                "metadata": [
                    "E-MTAB-552.rnaseq.design.csv:md5,142d6bb20d784ba414fa76a6e73bb37f"
                ],
                "normalized": [
                    
                ],
                "raw": [
                    "E-MTAB-552.rnaseq.raw.csv:md5,315f6ab694cb809bd33f481ad326c5a7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-17T13:31:14.414421877"
    },
    "Transcriptome Analysis of the potato (rnaseq)": {
        "content": [
            {
                "0": [
                    [
                        "E-MTAB-552",
                        "E_MTAB_552_rnaseq.raw.csv:md5,830f50b60b17b62f9ca2f6a163a2879f"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "E_MTAB_552_rnaseq.design.csv:md5,39c277eb6c24d7753f934be0cb743472"
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "R",
                        "[1] \"R version 4.3.3 (2024-02-29)\""
                    ]
                ],
                "4": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "ExpressionAtlas",
                        "[1] \u20181.30.0\u2019"
                    ]
                ],
                "design": [
                    "E_MTAB_552_rnaseq.design.csv:md5,39c277eb6c24d7753f934be0cb743472"
                ],
                "normalized": [
                    
                ],
                "raw": [
                    [
                        "E-MTAB-552",
                        "E_MTAB_552_rnaseq.raw.csv:md5,830f50b60b17b62f9ca2f6a163a2879f"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-27T20:34:29.182541795"
    },
    "Transcription profiling by array of Arabidopsis mutant for fis2": {
        "content": [
            {
                "0": [
                    "E-TABM-1007.A-AFFY-2.design.csv:md5,cb094d8a7f92d660a9d88d6eb6ec704a"
                ],
                "1": [
                    
                ],
                "2": [
                    "E-TABM-1007.A-AFFY-2.normalized.csv:md5,53672c5a95c252dadcf9ba718bdf2866"
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "R",
                        "[1] \"R version 4.3.3 (2024-02-29)\""
                    ]
                ],
                "4": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "ExpressionAtlas",
                        "[1] \u20181.30.0\u2019"
                    ]
                ],
                "metadata": [
                    "E-TABM-1007.A-AFFY-2.design.csv:md5,cb094d8a7f92d660a9d88d6eb6ec704a"
                ],
                "normalized": [
                    "E-TABM-1007.A-AFFY-2.normalized.csv:md5,53672c5a95c252dadcf9ba718bdf2866"
                ],
                "raw": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-17T13:28:23.604650949"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "E-MTAB-552_rnaseq.csv:md5,315f6ab694cb809bd33f481ad326c5a7"
                ],
                "csv": [
                    "E-MTAB-552_rnaseq.csv:md5,315f6ab694cb809bd33f481ad326c5a7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-14T18:28:01.099092539"
    },
    "Transcription profiling by array of Arabidopsis mutant for fis2 (microarray)": {
        "content": [
            {
                "0": [
                    [
                        "E-TABM-1007",
                        "E-TABM-1007.A-AFFY-2.design.csv:md5,cb094d8a7f92d660a9d88d6eb6ec704a"
                    ]
                ],
                "1": [
                    [
                        "E-TABM-1007",
                        [
                            "E-TABM-1007.A-AFFY-2.design.csv:md5,cb094d8a7f92d660a9d88d6eb6ec704a",
                            "E-TABM-1007.A-AFFY-2.normalized.csv:md5,53672c5a95c252dadcf9ba718bdf2866"
                        ]
                    ]
                ],
                "2": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "R",
                        "[1] \"R version 4.3.3 (2024-02-29)\""
                    ]
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETDATA",
                        "ExpressionAtlas",
                        "[1] \u20181.30.0\u2019"
                    ]
                ],
                "csv": [
                    [
                        "E-TABM-1007",
                        [
                            "E-TABM-1007.A-AFFY-2.design.csv:md5,cb094d8a7f92d660a9d88d6eb6ec704a",
                            "E-TABM-1007.A-AFFY-2.normalized.csv:md5,53672c5a95c252dadcf9ba718bdf2866"
                        ]
                    ]
                ],
                "design": [
                    [
                        "E-TABM-1007",
                        "E-TABM-1007.A-AFFY-2.design.csv:md5,cb094d8a7f92d660a9d88d6eb6ec704a"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-17T22:51:22.039416279"
    }
}